### PR TITLE
Signup: Update DesignTypeWithAtomicStoreStep to use TileGrid

### PR DESCRIPTION
**Pull request by tyxla**
This PR updates `DesignTypeWithAtomicStoreStep` to use the `TileGrid` component family for displaying all the tiles. It also reuses some classes from `DesignTypeWithStoreStep` - we can later move these to a more common location, so they can be reused. 

A change that's worth mentioning is that it will now rely on `this.state.showStore` directly for showing/hiding the atomic store, vs. unnecessarily using a className for that.

This allows to save a bunch of CSS and markup that we're reusing in other design type steps in Signup and Jetpack Onboarding. Can be followed by a couple more cleanup PRs after merging.

Contains #20640, which fixes a small visual bug with card link indicators on mobile devices.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/start/design-type-with-store-nux
* Try both different variants of the `signupAtomicStoreVsPressable` AB test.
* Verify there are no visual/functional differences when comparing to production.
* Verify the step works like it did before.

### CODE REVIEW

File: client/components/tile-grid/style.scss [hunk 1]
```diff
@@ -90,6 +94,7 @@
 			border-top: 2px solid lighten( $gray, 20% );
 			border-right: 2px solid lighten( $gray, 20% );
 			transform: rotate(45deg);
+			color: #fff;
```
Comment (MichaelArestad) [inline client/components/tile-grid/style.scss]: Can we make this `$white`

File: client/signup/steps/design-type-with-atomic-store/style.scss [hunk 2]
```diff
@@ -1,193 +0,0 @@
-.design-type-with-atomic-store {
```
Comment (MichaelArestad) [inline client/signup/steps/design-type-with-atomic-store/style.scss]: I love all this red

File: client/components/tile-grid/style.scss (hunk 1 above)
Comment (tyxla) [reply client/components/tile-grid/style.scss]: 🤦‍♂️

File: client/signup/steps/design-type-with-atomic-store/style.scss (hunk 2 above)
Comment (lamosty) [reply client/signup/steps/design-type-with-atomic-store/style.scss]: First we copy it and then we delete it 💯

File: client/components/tile-grid/style.scss (hunk 1 above)
Comment (tyxla) [reply client/components/tile-grid/style.scss]: Fixed in 767901dbaa7d15a800ddb58384cd9d08744af165.

**Comment by tyxla** [comment]
This one is ready to go from my perspective; could anyone please test and code-review it? Thanks!

**Comment by tyxla** [comment]
Good catch @sirreal! It seems there were visual issues showing the Pressable store.

However, I've reverted the approach a bit, because after fixing the visual issue that was preventing the Pressable store from appearing, the previous approach was erasing animations between steps. 

I've updated the PR, feel free to have another look and test it again. Thanks!

**Comment by sirreal** [comment]
I tested this and it seems to work well now. I'll give it a thorough review and look the code over tomorrow.

File: client/signup/steps/design-type-with-atomic-store/index.jsx [hunk 3]
```diff
@@ -12,15 +12,10 @@ import { includes, invoke } from 'lodash';
...
+import Tile from 'components/tile-grid/tile';
+import TileGrid from 'components/tile-grid';
 import { localize } from 'i18n-calypso';
 import { recordTracksEvent } from 'state/analytics/actions';
-import {
-	BlogImage,
-	PageImage,
-	GridImage,
-	StoreImage,
-} from '../design-type-with-atomic-store/type-images';
```
Comment (sirreal) [inline client/signup/steps/design-type-with-atomic-store/index.jsx]: Can we remove the entire file these are imported from?

https://github.com/Automattic/wp-calypso/blob/e9a8179b5a30238927d65c499fe5357018e99645/client/signup/steps/design-type-with-atomic-store/type-images.jsx#L1-L23

File: client/signup/steps/design-type-with-atomic-store/index.jsx [hunk 4]
```diff
@@ -187,9 +176,13 @@ class DesignTypeWithAtomicStoreStep extends Component {
 					/>
 				</div>
 				<div className={ designTypeListClassName }>
-					{ this.getChoices().map( this.renderChoice ) }
+					<TileGrid className="design-type-with-atomic-store__list">
```
Comment (sirreal) [inline client/signup/steps/design-type-with-atomic-store/index.jsx]: Look like this `className` can be removed.

File: client/signup/steps/design-type-with-atomic-store/index.jsx [hunk 5]
```diff
@@ -172,12 +161,12 @@ class DesignTypeWithAtomicStoreStep extends Component {
...
 		} );
 
-		const designTypeListClassName = classNames( 'design-type-with-atomic-store__list', {
+		const designTypeListClassName = classNames( 'design-type-with-store__list', {
 			'is-hidden': this.state.showStore,
 		} );
 
 		return (
-			<div className="design-type-with-atomic-store__substep-wrapper">
+			<div className="design-type-with-atomic-store__substep-wrapper design-type-with-store__substep-wrapper">
```
Comment (sirreal) [inline client/signup/steps/design-type-with-atomic-store/index.jsx]: Both classNames are needed for namespace lint rule?

File: client/signup/steps/design-type-with-atomic-store/index.jsx (hunk 3 above)
Comment (tyxla) [reply client/signup/steps/design-type-with-atomic-store/index.jsx]: 👍  8ce9473bd06dee6874984db661f55d1f60bab94f

File: client/signup/steps/design-type-with-atomic-store/index.jsx (hunk 5 above)
Comment (tyxla) [reply client/signup/steps/design-type-with-atomic-store/index.jsx]: The first one is necessary for the lint rule, the second one - for making the container relative. Can be cleaned up after we migrate the rest of the steps to use `TileGrid`.

File: client/signup/steps/design-type-with-atomic-store/index.jsx (hunk 4 above)
Comment (tyxla) [reply client/signup/steps/design-type-with-atomic-store/index.jsx]: 👍  2eedfce893f6ca32fc4db13fb0f48121159b32c7

**Comment by tyxla** [comment]
Thank you all for reviewing/testing 👍

**Pull request merged**